apps/spruce/src/components/Spawn/utils/hostUptime.ts

```diff
@@ -0,0 +1,247 @@
+    enabledWeekdays,
+    stopTime,
+    runContinuously,
+    startTime,
+  });
+
+  if (useDefaultUptimeSchedule) {
+    return {
+      enabledHoursCount,
+      errors: [],
+      warnings: [],
+    };
+  }
+
+  if (enabledHoursCount > maxUptimeHours) {
+    // Return error based on whether runContinously enabled
+    if (runContinuously) {
+      return {
+        enabledHoursCount,
+        errors: ["Please pause your host for at least 1 day per week."],
+        warnings: [],
+      };
+    }
+    const hourlyRequirement = Math.floor(maxUptimeHours / enabledWeekdaysCount);
+    return {
+      enabledHoursCount,
+      errors: [
+        `Please reduce your host uptime to a max of ${hourlyRequirement} hours per day.`,
+      ],
+      warnings: [],
+    };
+  }
+
+  if (enabledHoursCount > suggestedUptimeHours) {
+    // Return warning based on whether runContinuously enabled
+    if (runContinuously) {
+      return {
+        enabledHoursCount,
+        errors: [],
+        warnings: ["Consider pausing your host for 2 days per week."],
+      };
+    }
+    const hourlySuggestion = Math.floor(
+      suggestedUptimeHours / enabledWeekdaysCount,
+    );
+    return {
+      enabledHoursCount,
+      errors: [],
+      warnings: [
+        `Consider running your host for ${hourlySuggestion} hours per day or fewer.`,
+      ],
+    };
+  }
+  // No error
+  return {
+    enabledHoursCount,
+    errors: [],
+    warnings: [],
+  };
+};
+
+export const getEnabledHoursCount = ({
+  enabledWeekdays,
+  runContinuously,
+  startTime,
+  stopTime,
+}: Omit<ValidateInput, "useDefaultUptimeSchedule">) => {
+  const enabledWeekdaysCount =
+    enabledWeekdays?.filter((day) => day).length ?? 0;
+  const enabledHoursCount = runContinuously
+    ? enabledWeekdaysCount * hoursInDay
+    : enabledWeekdaysCount * getDailyUptime({ startTime, stopTime });
+  return { enabledHoursCount, enabledWeekdaysCount };
+};
+
+export const getSleepSchedule = (
+  { sleepSchedule, useDefaultUptimeSchedule }: HostUptime,
+  timeZone: string,
+): SleepScheduleInput => {
+  if (useDefaultUptimeSchedule) {
+    return {
+      ...defaultSleepSchedule,
+      timeZone,
+    };
+  }
+
+  const {
+    enabledWeekdays,
+    timeSelection: { runContinuously, startTime, stopTime },
+  } = sleepSchedule;
+
+  return {
+    dailyStartTime: runContinuously ? "" : toTimeString(new Date(startTime)),
+    dailyStopTime: runContinuously ? "" : toTimeString(new Date(stopTime)),
+    permanentlyExempt: false,
+    temporarilyExemptUntil: null,
+    timeZone,
+    shouldKeepOff: false,
+    wholeWeekdaysOff: enabledWeekdays.reduce((accum, isEnabled, i) => {
+      if (!isEnabled) {
+        accum.push(i);
+      }
+      return accum;
+    }, []),
+  };
+};
+
+const getDailyUptime = ({ startTime, stopTime }) =>
+  differenceInHours(new Date(stopTime), new Date(startTime));
+
+const toTimeString = (date: Date): string =>
+  date.toLocaleTimeString([], {
```

Comment:
toTimeString could possibly fit in the `utils/string` file

Reply:
Because of the way JS `Date` functions handle locales (the `[]` part of the function call) I am wary of genericizing this at the moment